frame/democracy/src/lib.rs

```diff
@@ -0,0 +1,2084 @@
+				for who in whos.into_iter() {
+					T::NativeCurrency::transfer_held(
+						&who,
+						&T::TreasuryAccount::get(),
+						amount,
+						true,
+						false,
+					)?;
+				}
+			}
+
+			Ok(())
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	// exposed immutables.
+
+	/// Get the amount locked in support of `proposal`; `None` if proposal isn't a valid proposal
+	/// index.
+	pub fn backing_for(proposal: PropIndex) -> Option<BalanceOf<T>> {
+		Self::deposit_of(proposal).map(|(l, d)| d * (l.len() as u32).into())
+	}
+
+	/// Get all referenda ready for tally at block `n`.
+	pub fn maturing_referenda_at(
+		n: T::BlockNumber,
+	) -> Vec<(ReferendumIndex, ReferendumStatus<T::BlockNumber, T::Hash, BalanceOf<T>, T::AssetId>)>
+	{
+		let next = Self::lowest_unbaked();
+		let last = Self::referendum_count();
+		Self::maturing_referenda_at_inner(n, next..last)
+	}
+
+	fn maturing_referenda_at_inner(
+		n: T::BlockNumber,
+		range: core::ops::Range<PropIndex>,
+	) -> Vec<(ReferendumIndex, ReferendumStatus<T::BlockNumber, T::Hash, BalanceOf<T>, T::AssetId>)>
+	{
+		range
+			.into_iter()
+			.map(|i| (i, Self::referendum_info(i)))
+			.filter_map(|(i, maybe_info)| match maybe_info {
+				Some(ReferendumInfo::Ongoing(status)) => Some((i, status)),
+				_ => None,
+			})
+			.filter(|(_, status)| status.end == n)
+			.collect()
+	}
+
+	// Exposed mutables.
+
+	/// Start a referendum.
+	pub fn internal_start_referendum(
+		proposal_hash: T::Hash,
+		asset_id: T::AssetId,
+		threshold: VoteThreshold,
+		delay: T::BlockNumber,
+	) -> ReferendumIndex {
+		<Pallet<T>>::inject_referendum(
+			<frame_system::Pallet<T>>::block_number() + T::VotingPeriod::get(),
+			ProposalId { hash: proposal_hash, asset_id },
+			threshold,
+			delay,
+		)
+	}
+
+	/// Remove a referendum.
+	pub fn internal_cancel_referendum(ref_index: ReferendumIndex) {
+		Self::deposit_event(Event::<T>::Cancelled(ref_index));
+		ReferendumInfoOf::<T>::remove(ref_index);
+	}
+
+	// private.
+
+	/// Ok if the given referendum is active, Err otherwise
+	fn ensure_ongoing(
+		r: ReferendumInfo<T::BlockNumber, T::Hash, BalanceOf<T>, T::AssetId>,
+	) -> Result<ReferendumStatus<T::BlockNumber, T::Hash, BalanceOf<T>, T::AssetId>, DispatchError>
+	{
+		match r {
+			ReferendumInfo::Ongoing(s) => Ok(s),
+			_ => Err(Error::<T>::ReferendumInvalid.into()),
+		}
+	}
+
+	fn referendum_status(
+		ref_index: ReferendumIndex,
+	) -> Result<ReferendumStatus<T::BlockNumber, T::Hash, BalanceOf<T>, T::AssetId>, DispatchError>
+	{
+		let info = ReferendumInfoOf::<T>::get(ref_index).ok_or(Error::<T>::ReferendumInvalid)?;
+		Self::ensure_ongoing(info)
+	}
+
+	/// Actually enact a vote, if legit.
+	fn try_vote(
+		who: &T::AccountId,
+		ref_index: ReferendumIndex,
+		vote: AccountVote<BalanceOf<T>>,
+	) -> DispatchResult {
+		let mut status = Self::referendum_status(ref_index)?;
+		ensure!(
+			vote.balance() <= T::Currency::free_balance(status.proposal_id.asset_id, who),
```

Comment:
could you rename T::Currency to be T::MultiCurreny?

Reply:
may be retain T::Currency as if was before to ease diff with original pallet

Reply:
sure, left as-is.